src/js_collection.hpp

```diff
@@ -48,8 +48,14 @@ typename T::Value CollectionClass<T>::create_collection_change_set(ContextType c
 {
     ObjectType object = Object::create_empty(ctx);
     std::vector<ValueType> deletions, insertions, modifications;
-    for (auto index : change_set.deletions.as_indexes()) {
-        deletions.push_back(Value::from_number(ctx, index));
+    
+    if (change_set.deletions.count() == std::numeric_limits<size_t>::max()) {
+        deletions.push_back(Value::from_undefined(ctx));
```

Comment:
I thought about that as well. Will asking him for input here. We should pick the most meaningful value here. I choose 'undefined' since the naming is more meaningful to the natural English language.  It kinda falls more to the result we are trying to communicate. The server knows there are deletions but their count is "undefined". Not sure this is a correct/best way to communicate this. I wanted to escape from things like -1 or Positive infinity or some other obscure value.